src/OrchardCore.Modules/OrchardCore.Flows/Settings/BagPartSettingsDisplayDriver.cs

```diff
@@ -44,22 +49,56 @@ public override async Task<IDisplayResult> UpdateAsync(ContentTypePartDefinition
         {
             var model = new BagPartSettingsViewModel();
 
-            await context.Updater.TryUpdateModelAsync(model, Prefix, m => m.ContainedContentTypes, m => m.DisplayType);
+            await context.Updater.TryUpdateModelAsync(model, Prefix, m => m.ContainedContentTypes, m => m.DisplayType, m => m.Source, m => m.Stereotypes);
 
-            if (model.ContainedContentTypes == null || model.ContainedContentTypes.Length == 0)
+            switch (model.Source)
             {
-                context.Updater.ModelState.AddModelError(nameof(model.ContainedContentTypes), S["At least one content type must be selected."]);
+                case BagPartSettingType.ContentTypes:
+                    SetContentTypes(context, model);
+                    break;
+                case BagPartSettingType.Stereotypes:
+                    SetStereoTypes(context, model);
+                    break;
+                default:
+                    context.Updater.ModelState.AddModelError(Prefix, nameof(model.Source), S["Content type source must be set with a valid value."]);
+                    break;
             }
-            else
+
+            return Edit(contentTypePartDefinition, context.Updater);
+        }
+
+        private void SetStereoTypes(UpdateTypePartEditorContext context, BagPartSettingsViewModel model)
+        {
+            if (String.IsNullOrEmpty(model.Stereotypes))
             {
-                context.Builder.WithSettings(new BagPartSettings
-                {
-                    ContainedContentTypes = model.ContainedContentTypes,
-                    DisplayType = model.DisplayType
-                });
+                context.Updater.ModelState.AddModelError(Prefix, nameof(model.Stereotypes), S["Please provide a Stereotype."]);
+
+                return;
             }
 
-            return Edit(contentTypePartDefinition, context.Updater);
+            context.Builder.WithSettings(new BagPartSettings
+            {
+                ContainedContentTypes = Array.Empty<string>(),
+                ContainedStereotypes = model.Stereotypes.Split(',', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries),
+                DisplayType = model.DisplayType
+            });
+        }
+
+        private void SetContentTypes(UpdateTypePartEditorContext context, BagPartSettingsViewModel model)
+        {
+            if (model.ContainedContentTypes == null || model.ContainedContentTypes.Length == 0)
```

Comment:
I think null `coalescing` operator could be used in many places in your PR

Reply:
I agree with you. I like using the null coalescing when it make sense. But I don’t like using it inside an of condition. Either way, it’s a preference

Reply:
If you look there are many places that you are checking both nullability and length while you can use the available C# operators for that

Reply:
@hishamco I am aware. As mentioned previously, it's a preference. Additionally, I don't think I ever saw something like this in OC code.

```
if ((model.ContainedContentTypes?.Length) ?? 0 == 0)
{

}
```
ugggggh, that hurt my eyes.

We could do something like this instead, but that is just a wast of memory allocation

```
var totalContainedTypes = model.ContainedContentTypes?.Length ?? 0;

if (totalContainedTypes  == 0)
{
}
```

Reply:
What about

```csharp
if (model.ContainedContentTypes?.Length == 0)
{

}
```

Reply:
It doesn't have the same behavior if `ContainedContentTypes` is null because you get a nullable int without any value, so it is neither == 0, nor != 0, nor >=0 and so on.

You would need to use the following which is not simpler.

    if (!(model.ContainedContentTypes?.Length).HasValue ||
        model.ContainedContentTypes.Length == 0)
    {

    }

I think the following would work but it is not so clear.

    if (!(model.ContainedContentTypes?.Length > 0))
    {

    }

Reply:
@hishamco as @jtkech mentioned that it's not the same. 

As you can see the the original code in the PR is not the shortest, but is the most readable.
```
if (model.ContainedContentTypes == null || model.ContainedContentTypes.Length == 0)
```

Reply:
```csharp
if (!(model.ContainedContentTypes?.Any() ?? false))
{
}
```

This should work ;) or we can keep as you did
```

Reply:
@hishamco we all can agree that this can be written in many different ways. I don't want to complicate the code for no good reason (other than making it too fancy :) ). I would rather leave it as is for simple readability

Reply:
Leave it as its, then I can refactor this later